yearn/contracts/ConvexStrategy.sol

```diff
@@ -0,0 +1,614 @@
+        if (crvBalance > 0) {
+            // Deposit a portion of CRV to the voter to gain CRV boost.
+            crvBalance = adjustCRV(crvBalance);
+
+            IERC20(crv).safeApprove(uniswap, 0);
+            IERC20(crv).safeApprove(uniswap, crvBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = crv;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                crvBalance,
+                uint256(0),
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Buy WBTC using obtained CVX tokens. Use SushiSwap as CVX is not
+        // supported by UniSwap.
+        uint256 cvxBalance = IERC20(cvx).balanceOf(address(this));
+        if (cvxBalance > 0) {
+            IERC20(cvx).safeApprove(sushiswap, 0);
+            IERC20(cvx).safeApprove(sushiswap, cvxBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = cvx;
+            path[1] = weth;
+            path[2] = wbtc;
+
+            IUniswapV2Router(sushiswap).swapExactTokensForTokens(
+                cvxBalance,
+                uint256(0),
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Buy WBTC using obtained extra reward tokens, if applicable.
+        if (tbtcConvexExtraReward != address(0)) {
+            uint256 extraRewardBalance = IERC20(tbtcConvexExtraReward)
+            .balanceOf(address(this));
+            if (extraRewardBalance > extraRewardSwapThreshold) {
+                IERC20(tbtcConvexExtraReward).safeApprove(uniswap, 0);
+                IERC20(tbtcConvexExtraReward).safeApprove(
+                    uniswap,
+                    extraRewardBalance
+                );
+
+                address[] memory path = new address[](3);
+                path[0] = tbtcConvexExtraReward;
+                path[1] = weth;
+                path[2] = wbtc;
+
+                IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                    extraRewardBalance,
+                    uint256(0),
+                    path,
+                    address(this),
+                    now
+                );
+            }
+        }
+
+        // Deposit acquired WBTC to the Curve pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtc).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtc).safeApprove(tbtcCurvePoolDepositor, 0);
+            IERC20(wbtc).safeApprove(tbtcCurvePoolDepositor, wbtcBalance);
+            ICurvePool(tbtcCurvePoolDepositor).add_liquidity(
+                [0, 0, wbtcBalance, 0],
+                0
+            );
+        }
+
+        // Calculate the profit after obtaining new vault's underlying tokens.
+        profit = want.balanceOf(address(this)).sub(initialWantBalance);
```

Comment:
I think we can call `balanceOfWant()` instead.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/10/commits/7d20ec730332e0af308f37dfeeebf06c0db2b998